src/sql/src/statements.rs

```diff
@@ -260,6 +260,15 @@ pub fn sql_value_to_value(
             (*b).into()
         }
         SqlValue::DoubleQuotedString(s) | SqlValue::SingleQuotedString(s) => {
+            // ensure data_type is not interval type when parsing string
+            ensure!(
+                !matches!(data_type, ConcreteDataType::Interval(_)),
+                ColumnTypeMismatchSnafu {
+                    column_name,
+                    expect: data_type.clone(),
+                    actual: ConcreteDataType::string_datatype(),
+                }
+            );
```

Comment:
This fix is unnecessary,looks like just fixing the `unreacable` is enough.

https://github.com/GreptimeTeam/greptimedb/blob/87bd12d6df4e0e7eeb75a8ad5996515ccd9fc049/src/sql/src/statements.rs#L140

Reply:
Do we still need this `ensure!` after fixing `unreachable`?

Reply:
will check tomorrow, not around the computer today.<